src/touchhandler/TouchHandler.h

```diff
@@ -3,42 +3,34 @@
 #include "displayapp/TouchEvents.h"
 
 namespace Pinetime {
-  namespace Components {
-    class LittleVgl;
-  }
-
-  namespace Drivers {
-    class Cst816S;
-  }
-
   namespace Controllers {
     class TouchHandler {
     public:
-      explicit TouchHandler(Drivers::Cst816S&, Components::LittleVgl&);
-      void CancelTap();
-      bool GetNewTouchInfo();
-      void UpdateLvglTouchPoint();
+      struct TouchPoint {
+        int x;
+        int y;
+        bool touching;
+      };
+
+      bool ProcessTouchInfo(Drivers::Cst816S::TouchInfos info);
 
       bool IsTouching() const {
-        return info.touching;
+        return currentTouchPoint.touching;
       }
 
       uint8_t GetX() const {
-        return info.x;
+        return currentTouchPoint.x;
       }
 
       uint8_t GetY() const {
-        return info.y;
+        return currentTouchPoint.y;
       }
 
       Pinetime::Applications::TouchEvents GestureGet();
 
     private:
-      Pinetime::Drivers::Cst816S::TouchInfos info;
-      Pinetime::Drivers::Cst816S& touchPanel;
-      Pinetime::Components::LittleVgl& lvgl;
       Pinetime::Applications::TouchEvents gesture;
-      bool isCancelled = false;
+      TouchPoint currentTouchPoint = {0};
```

Comment:
without a constructor what is the contents of the initial `currentTouchPoint`. It feels for me as it is `x=0` and `y=undefined` and `touching=undefined`

Reply:
I think they get implicitly initialized to zero, but explicitly initializing only the first variable is misleading indeed.